### PR TITLE
Switch to using the networkMonitor and EventLink settings in the db

### LINKED ISSUE #4769: Move network monitor and event link prefs to the database
Steps to follow (in this order):
- [x] Add the `enable_network_monitor` and `enable_event_link` to the `user_settings` table. Default to `false`.
- [x] Add the new settings to the `schema.ts` and `models.ts`.
  - (https://github.com/RecordReplay/backend/pull/3900)
- [x] Connect the new `user_settings` field to the frontend.
  - (https://github.com/RecordReplay/devtools/pull/4770)
- [x] Remove the old local features flag for `network` and `eventLink`.
  - (https://github.com/RecordReplay/devtools/pull/4770)

### PR DESCRIPTION
**Pull request by jaril**
Fix #4769.

### CODE REVIEW

File: src/ui/components/Events/Events.tsx [hunk 1]
```diff
@@ -38,7 +41,12 @@ function Events({ currentTime, events, executionPoint, seek }: PropsFromRedux) {
           <div key={e.point}>
             <CurrentTimeLine isActive={currentEventIndex === i} />
             <div className="px-1.5">
-              <Event onSeek={onSeek} event={e} {...{ currentTime, executionPoint }} />
+              <Event
+                onSeek={onSeek}
+                event={e}
+                {...{ currentTime, executionPoint }}
```
Comment (jcmorrow) [inline src/ui/components/Events/Events.tsx:47]: I still don't know how I feel about this style, but if we're going this route you should include `onSeek`

File: src/ui/graphql/settings.ts [hunk 2]
```diff
@@ -18,6 +18,8 @@ export const GET_USER_SETTINGS = gql`
         enableRepaint
         enableGlobalSearch
         disableLogRocket
+        enableNetworkMonitor
+        enableEventLink
```
Comment (jcmorrow) [inline src/ui/graphql/settings.ts:22]: sort these maybe?

File: src/ui/hooks/settings.ts [hunk 3]
```diff
@@ -19,6 +19,8 @@ const emptySettings: UserSettings = {
   enableGlobalSearch: false,
   defaultWorkspaceId: null,
   disableLogRocket: false,
+  enableNetworkMonitor: false,
+  enableEventLink: false,
```
Comment (jcmorrow) [inline src/ui/hooks/settings.ts:23]: sort?

File: src/ui/hooks/settings.ts [hunk 4]
```diff
@@ -30,6 +32,8 @@ const testSettings: UserSettings = {
...
```
Comment (jcmorrow) [inline src/ui/hooks/settings.ts:36]: sort?

File: src/ui/hooks/settings.ts [hunk 5]
```diff
@@ -97,6 +101,8 @@ function convertUserSettings(data: any): UserSettings {
     enableGlobalSearch: settings.enableGlobalSearch,
     defaultWorkspaceId: data.viewer.defaultWorkspace?.id || null,
     disableLogRocket: settings.disableLogRocket,
+    enableNetworkMonitor: settings.enableNetworkMonitor,
+    enableEventLink: settings.enableEventLink,
```
Comment (jcmorrow) [inline src/ui/hooks/settings.ts:105]: sort?

File: src/ui/types/index.ts [hunk 6]
```diff
@@ -14,11 +14,11 @@ export type UserSettings = {
   enableTeams: boolean;
   showElements: boolean;
   showReact: boolean;
+  enableNetworkMonitor: boolean;
+  enableEventLink: boolean;
```
Comment (jcmorrow) [inline src/ui/types/index.ts:18]: sort?

File: src/ui/hooks/settings.ts (hunk 5 above)
Comment (jcmorrow) [inline src/ui/hooks/settings.ts:105]: sort?

File: src/ui/components/Events/Event.tsx [hunk 7]
```diff
@@ -43,7 +43,13 @@ const getEventLabel = (event: ReplayEvent) => {
   return label;
 };
 
-export default function Event({ onSeek, currentTime, executionPoint, event }: EventProps) {
+export default function Event({
+  onSeek,
+  currentTime,
+  executionPoint,
+  event,
+  showLink,
```
Comment (jcmorrow) [inline src/ui/components/Events/Event.tsx:51]: sort?

File: src/ui/components/Events/Event.tsx [hunk 8]
```diff
@@ -4,13 +4,13 @@ import { ReplayEvent } from "ui/state/app";
...
 import MaterialIcon from "../shared/MaterialIcon";
 import Matches from "./Matches";
-import { features } from "ui/utils/prefs";
 
 type EventProps = {
   event: ReplayEvent;
   currentTime: any;
   executionPoint: any;
   onSeek: (point: string, time: number) => void;
+  showLink: boolean;
```
Comment (jcmorrow) [inline src/ui/components/Events/Event.tsx:13]: sort?

**Review by jcmorrow — APPROVED** [review]
Sorry I left so many sort comments, I know this seems pedantic when you first start but it honestly does save time and avoid bugs in the long run.

File: src/ui/components/Events/Events.tsx (hunk 1 above)
Comment (jaril) [reply src/ui/components/Events/Events.tsx:47]: Not a fan anymore — I'm undoing it as I go!

**Comment by jaril** [comment]
There's type errors that appear to be unrelated to this patch.

**Pull request merged**